File: go/validate-stack-sequences.go
func validateStackSequences(pushed []int, popped []int) bool {
	s := make([]int, 0)
	j := 0
	for i := 0; i < len(pushed); i++ {
		s = append(s, pushed[i])
		for 0 < len(s) && s[len(s)-1] == popped[j] {
			s = s[:len(s)-1]
			j += 1
		}
	}
	return len(s) == 0
}

File: python/validate-stack-sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        s = []
        i, j = 0, 0
        while i < len(pushed):
            s.append(pushed[i])
            while s and s[-1] == popped[j]:
                s.pop()
                j += 1
            i += 1
        return len(s) == 0
